charts/connectivity-monitor/templates/ds.yaml

```diff
@@ -0,0 +1,68 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: connectivity-exporter-config
+  namespace: {{ .Values.namespace }}
+  labels: {app: connectivity-exporter}
+data:
+  metrics-addr: {{ .Values.metricsAddr }}
+  filtered-ports: "443,6443"
+  filtered-ips: "0.0.0.0/0"
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: connectivity-exporter
+  namespace: {{ .Values.namespace }}
+  labels: {app: connectivity-exporter}
+spec:
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 100%
+
+  selector: {matchLabels: {app: connectivity-exporter}}
+  template:
+    metadata: {labels: {app: connectivity-exporter}}
+    spec:
+      serviceAccountName: connectivity-monitor
+      hostNetwork: true
+      priorityClassName: connectivity-exporter
+      dnsPolicy: ClusterFirstWithHostNet
+
+      containers:
+
+      - name: connectivity-exporter
+        image: {{ .Values.image.registry}}/{{ .Values.image.name }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+        - /bin/sh
+        - -c
+        - |
+          /bin/connectivity-exporter -r $FILTERED_IPS -p $FILTERED_PORTS -v 0 -metrics-addr $METRICS_ADDR
+          echo "Good bye!"
+        env:
+          - name: METRICS_ADDR
+            valueFrom:
+              configMapKeyRef:
+                name: connectivity-exporter-config
+                key: metrics-addr
```

Comment:
I removed the environment variable `METRICS_ADDR` since it was not working with the `:` in the `ports` and `readinessProbe` section.

Otherwise for `$FILTERED_IPS` and `$FILTERED_PORTS`, I think it is different. Some Helm values could be set at build time, but `$FILTERED_IPS` could be changed at run-time, so having them in a ConfigMap could be useful. When it is installed on the seed, the connectivity-exporter process could watch the ConfigMap with the Kubernetes API and automatically change the BPF map accordingly, without needing to restart the DaemonSet when a new shoot is deployed.